byzcoin/service.go

```diff
@@ -1951,10 +1964,33 @@ func (s *Service) processOneTx(sst *stagingStateTrie, tx ClientTransaction) (Sta
 // GetContractConstructor gets the contract constructor of the contract
 // contractName.
 func (s *Service) GetContractConstructor(contractName string) (ContractFn, bool) {
-	fn, exists := s.contracts[contractName]
+	fn, exists := s.contracts.Search(contractName)
 	return fn, exists
 }
 
+// GetContract creates a contract given the ID and the bytes input if the contract
+// exists or nil otherwise.
+func (s *Service) GetContract(contractName string, in []byte) (Contract, error) {
```

Comment:
In fact a contract is an abstraction, what we return here is an instance of a contract: maybe `getInstance()`, or `getContractInstance()` ?